.github/actions/download-cantera-test/action.yml

```diff
@@ -0,0 +1,50 @@
+name: "Download Cantera Test Files"
+description: "Download and cache Cantera's Python test suite"
+inputs:
+  incoming-sha:
+    description: The hash of the commit that should be downloaded
+    required: true
+outputs:
+  test-root:
+    description: The root folder where the tests are located
+    value: ${{ steps.set-output.outputs.DESTINATION_PATH }}
+
+runs:
+  using: "composite"
+  steps:
+  - name: Sanitize the destination path
+    run: |
+      $DESTINATION_PATH = "${{ runner.temp }}" -replace "\\", "/"
+      echo "DESTINATION_PATH=$DESTINATION_PATH" | Out-File -FilePath $Env:GITHUB_ENV -Encoding utf8 -Append
+    if: runner.os == 'Windows'
+    shell: pwsh
+  - name: Sanitize the destination path
+    run: echo "DESTINATION_PATH=${{ runner.temp }}" >> $GITHUB_ENV
+    shell: bash
+    if: runner.os != 'Windows'
+  - name: Set the outputs
+    id: set-output
+    run: echo "DESTINATION_PATH=${{ env.DESTINATION_PATH }}" >> $GITHUB_OUTPUT
+    shell: bash
+  - name: Restore cached the test files
+    uses: actions/cache/restore@v4
+    id: restore-cache
+    with:
+      path: ${{ steps.set-output.outputs.DESTINATION_PATH }}/test
+      # Bump the integer to bust the cache
+      key: ${{ inputs.incoming-sha }}-0
```

Comment:
Yeah, makes sense. The warnings are spurious since the cache is the same for all runners and it's a race, but if it's that fast it's really not needed.